### ISSUE TITLE
A2Q2

### ISSUE THREAD

**Issue by rennmeli** [BME1478H/Winter2020class#12]
Having issues with my code. It must not be reading it properly because 20180405-01 doesn't have a notes file yet it is saying that it does. 

![Screen Shot 2020-01-15 at 11 00 33 PM](https://user-images.githubusercontent.com/59658814/72492394-0c371480-37eb-11ea-8857-b2abadf2154e.png)

cd documents/bme1478/week_2/assignment2_data/assignment-data
FILES=$(ls)
for f in $FILES; do 
cd $f 
if find . -name note
then echo “$f not missing note file”
else echo “$f missing not file”
fi
cd ..
done

**Comment by RayNele** [comment]
so there's a couple problems with your if statement in the loop.

first of all, in bash, an if statement must be followed by a boolean in square brackets and terminated with a semicolon, and followed with a `then`

`if [ 'foo' = 'foo' ]; then`

So in the case of your example `find . -name note` is:

1) not enclosed in brackets
2) not terminated with a semicolon
3) not a boolean

a boolean can only have two values: `True` or `False`

An if statement can only function with a boolean for a simple reason. If we read the statement back into plain english, it means "if (the following expression is true) then (execute the following commands)". What you mean to do with your code is "if you find a a file with the name note, then (do something)."

However, when we evaluate `find . -name note` it does not return true or false, it returns the name of any file which matches the pattern `note`. Which in this case is actually nothing which brings up another problem.

4) `find` with the `-name` option actually matches EXACT filenames. So since there is no exact file with name "note" it will return nothing. We can get around this by using the wildcard `*note*` which indicates a pattern with any characters before or after the word "note". Or even better, we can match with the pattern `*.txt`which will match any text file regardless of whether it has "note" in the name.

So now that we've fixed problems 1,2, and 4, we can come back to 3. A more "creative" solution to continue using `find` in our boolean would be to pipe the output of `find . -name '*.txt'` into a the  `wc -l` (which does a line count) to return a number, which we can then compare with an integer number.

Basically, we're using the `find` and `wc` functions to not simply list the name of the file, but HOW MANY, which would either be 1 or 0 in this case of note.txt files.

After all of this, we would be simply comparing two numbers, which can either return `True` or `False`, which is exactly the definition of a boolean. 

This way we have essentially 

`if [ 1 = 1 ]; then`, which evaluates as true and results in `echo “$f not missing note file”`

and 

`if [ 0 = 1 ]; then` which evaluates as false and results in `else echo “$f missing not file”`

Just keep in mind the spaces between the `[]`. Bash is picky.

**Comment by rennmeli** [comment]
so now my code prints out the 0 and 1 but is not printing "$f missing note file": 
cd documents/bme1478/week_2/assignment2_data/assignment-data
FILES=$(ls)
for f in $FILES; do 
cd $f 
find . -name ‘*.txt’  | wc -l
if [1 = l ]; then echo “$f not missing note file”
else echo “$f missing note file”
fi
cd ..
done 
![Screen Shot 2020-01-16 at 10 46 01 AM](https://user-images.githubusercontent.com/59658814/72539605-88624400-384d-11ea-868d-41447151c807.png)

**Comment by rennmeli** [comment]
so the find command only works when it is outside the IF loop

**Comment by RayNele** [comment]
Sorry my explanation wasn't clear enough, I didn't mean to literally `if [ 1 = 1 ]; then`

the `find . -name ‘*.txt’ | wc -l` needs to be part of the if statement. you want to compare what that outputs to a 1, **not an L**, since when `find . -name ‘*.txt’ | wc -l`  is a 1, then that means a note file exists. 

you will probably need to assign `find . -name ‘*.txt’ | wc -l`  to a variable compare it to 1 in the if statement.

**Comment by rennmeli** [comment]
Is there anyone available to help in person? If so, how would I arrange it? 

cd documents/bme1478/week_2/assignment2_data/assignment-data
FILES=$(ls)
for f in $FILES; do 
cd $f 
$myvar= find . -name ‘*.txt’ | wc -l 
if [ $myvar= 1 ]; then echo “$f not missing note file”
else echo “$f missing note file”
fi
cd ..
done
